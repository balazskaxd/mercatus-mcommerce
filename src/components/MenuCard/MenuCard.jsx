import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  menuCard: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: 130,
    width: 130,
    margin: 12,
    borderWidth: 1,
    borderColor: 'white',
  },
  title: {
    color: 'white',
    fontSize: 22,
  },
});

const MenuCard = ({ title, navigate }) => (
  <TouchableOpacity style={styles.menuCard} onPress={() => navigate()}>
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
);

MenuCard.propTypes = {
  title: PropTypes.string.isRequired,
  navigate: PropTypes.func.isRequired,
};

export default MenuCard;
