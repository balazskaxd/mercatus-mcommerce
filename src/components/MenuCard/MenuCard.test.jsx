import { Text, TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import MenuCard from './MenuCard';


describe('MenuCard', () => {
  const props = {
    title: 'Title',
    navigate: jest.fn(),
  };

  it('MenuCard renders correctly', () => {
    renderer.create(
      <MenuCard title={props.title} navigate={props.navigate} />,
    );
  });

  it('renders the correct text', () => {
    const wrapper = renderer.create(<MenuCard title={props.title} navigate={props.navigate} />);

    const textInst = wrapper.root.findByType(Text);
    expect(textInst.props.children).toBe(props.title);
  });

  it('test onPress functionality', () => {
    const wrapper = renderer.create(<MenuCard title={props.title} navigate={props.navigate} />);

    wrapper.root.findByType(TouchableOpacity).props.onPress();
    expect(props.navigate).toHaveBeenCalledTimes(1);
  });
});
