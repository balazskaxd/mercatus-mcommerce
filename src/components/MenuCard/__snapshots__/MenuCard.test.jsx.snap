// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCard MenuCard renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "white",
      "borderWidth": 1,
      "display": "flex",
      "height": 130,
      "justifyContent": "center",
      "margin": 12,
      "opacity": 1,
      "width": 130,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 22,
      }
    }
  >
    Title
  </Text>
</View>
`;
